src/components/Contact/ContactForm.jsx

```diff
@@ -0,0 +1,144 @@
+import React, { useState } from "react";
+import styles from "./ContactForm.module.css";
+import FormElement from "../utility/FormElement/FormElement";
+import Button from "../utility/Button/Button";
+
+const ContactForm = () => {
+  const [formData, setFormData] = useState({
+    name: { value: "", error: "", touched: false },
+    email: { value: "", error: "", touched: false },
+    companyName: { value: "", error: "", touched: false },
+    title: { value: "", error: "", touched: false },
+    message: { value: "", error: "", touched: false },
+  });
+
+  const handleChange = (inputField) => (e) => {
+    const { value } = e.target;
+    setFormData((prevState) => ({
+      ...prevState,
+      [inputField]: {
+        ...prevState[inputField],
+        value: value,
+        error: "",
+        touched: true,
+      },
+    }));
+  };
+
+  // check if email is valid
+  const isEmailValid = (email) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    return emailRegex.test(email);
+  };
+
+  const handleBlur = (inputField) => () => {
+    if (!formData[inputField].value) {
+      setFormData((prevState) => ({
+        ...prevState,
+        [inputField]: {
+          ...prevState[inputField],
+          error: "This field is required",
+        },
+      }));
+    } else if (
+      inputField === "message" &&
+      formData[inputField].value.length < 25
+    ) {
+      setFormData((prevState) => ({
+        ...prevState,
+        [inputField]: {
+          ...prevState[inputField],
+          error: "Message must be at least 25 characters",
+        },
+      }));
+    } else if (
+      inputField === "email" &&
+      !isEmailValid(formData[inputField].value)
+    ) {
+      setFormData((prevState) => ({
+        ...prevState,
+        [inputField]: {
+          ...prevState[inputField],
+          error: "Invalid email address",
+        },
+      }));
+    }
+  };
+  const handleFocus = (inputField) => () => {
+    setFormData((prevState) => ({
+      ...prevState,
+      [inputField]: { ...prevState[inputField], error: "", touched: true },
+    }));
+  };
+  const isFormValid = () => {
+    return !Object.values(formData).some(
+      (field) => field.error || !field.touched
+    );
+  };
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setFormData({
+      name: { value: "", error: "", touched: false },
+      email: { value: "", error: "", touched: false },
+      companyName: { value: "", error: "", touched: false },
+      title: { value: "", error: "", touched: false },
+      message: { value: "", error: "", touched: false },
+    });
+  };
+  return (
+    <form className={styles.ContactForm}>
+      <div className={styles.ContactFormInputs}>
+        <FormElement
```

Comment:
We can reduce this further using map operator.

Reply:
Done.